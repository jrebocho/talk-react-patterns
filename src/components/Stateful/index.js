import React, { Component } from 'react'
import './Stateful.css'

class Stateful extends Component {
  constructor(props) {
    super(props)
    this.state = { on: false }
    this.toggle = this.toggle.bind(this)
  }

  toggle() {
    this.setState(prevState => ({
      on: !prevState.on
    }))
  }

  render() {
    const boxColor = this.state.on ? 'green' : 'red'

    return (
      <div onClick={this.toggle}>
        <div className={`box ${boxColor}`}>{this.state.on ? 'ON' : 'OFF'}</div>
      </div>
    )
  }
}

export default Stateful
