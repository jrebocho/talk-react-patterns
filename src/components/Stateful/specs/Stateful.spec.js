import React from 'react'
import { shallow } from 'enzyme'
import Stateful from '../../Stateful'

describe('<Stateful />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Stateful />)
    expect(wrapper).toMatchSnapshot()
  })

  describe('when the component is on', () => {
    it('renders correctly', () => {
      const wrapper = shallow(<Stateful />)
      wrapper.setState({ on: true })

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when the component is clicked', () => {
    it('renders correctly', () => {
      const wrapper = shallow(<Stateful />)
      wrapper.simulate('click')

      expect(wrapper).toMatchSnapshot()
    })
  })
})
