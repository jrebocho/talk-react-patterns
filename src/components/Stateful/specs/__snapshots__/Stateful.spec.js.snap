// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stateful /> renders correctly 1`] = `
<div
  onClick={[Function]}
>
  <div
    className="box red"
  >
    OFF
  </div>
</div>
`;

exports[`<Stateful /> when the component is clicked renders correctly 1`] = `
<div
  onClick={[Function]}
>
  <div
    className="box green"
  >
    ON
  </div>
</div>
`;

exports[`<Stateful /> when the component is on renders correctly 1`] = `
<div
  onClick={[Function]}
>
  <div
    className="box green"
  >
    ON
  </div>
</div>
`;
